### PR TITLE
Structurize definitions

### PR DESCRIPTION
**Pull request by simonvadee**
I introduce a new way to structurize definitions, which I found to work better using a recursive function to analyze the paths.
In this PR, I focused my work on bootstrapping the structuredefinitions, not really on the Attribute class because I don't really know how it will be integrated in the back/front yet, so I probably miss some features.
The `Definition` objects will be stored in redis and used in the graphql API when dealing with definitions. when serializing the definitions to JSON, the parent is omitted because having both children and parent would cause a gigantic mess.

Do you understand what's going on here of here is it too cryptic ?

### CODE REVIEW

File: src/definition.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  return slice.path === attr.path && !!slice.sliceName
+}
+
+export const isChildOf = (
+  child: AttributeDefinition,
+  parent: AttributeDefinition,
+): boolean => {
+  const parentPath = parent.path.split('.').slice(1)
+  const attrPath = child.path.split('.').slice(1)
+  return JSON.stringify(attrPath.slice(0, -1)) === JSON.stringify(parentPath)
```
[the remark this answers is not in the document]
Comment (Jasopaum) [reply src/definition.ts]: Also, do you need the .slice(1) above? The first element should be the same for parent and child, am I wrong?

[the remark this answers is not in the document]
Comment (simonvadee) [reply src/definition.ts]: you're right, I don't need the slice(1)

[the remark this answers is not in the document]
Comment (simonvadee) [reply src/definition.ts]: comparing the JSON representation of the two arrays is sadly the best way to check for array equality in javascript

[the remark this answers is not in the document]
Comment (Jasopaum) [reply src/definition.ts]: What about keeping strings and doing something like
```
parent.path === child.path.substring(0, child.path.lastIndexOf(".") )
```

[the remark this answers is not in the document]
Comment (simonvadee) [reply src/definition.ts]: that'd be simpler indeed! done

**Pull request merged**